### DISCUSSION TITLE
Installation Failure with Go Version Below 1.22

### DISCUSSION THREAD

**Discussion by smrrazavian** [projectdiscovery/dnsx#574] [General]
## Issue Description:
I faced a familiar challenge while trying to install dnsx and nuclei, akin to the issue detailed in [issue #1055](https://github.com/projectdiscovery/subfinder/issues/1055). Upon encountering installation failures, I opted to upgrade my Go version from 1.21 to 1.22, successfully resolving the problem.

## Suggested Action:
I propose updating the installation instructions in the [README.md](https://github.com/projectdiscovery/dnsx?tab=readme-ov-file#installation-instructions) file to provide clearer and more comprehensive documentation. This update should include specific information regarding compatibility with Go version 1.22, ensuring that users are adequately informed about the requirements for successful installation.

**Comment by ehsandeep** [comment]
@smrrazavian https://github.com/projectdiscovery/dnsx?tab=readme-ov-file#installation-instructions says:

> dnsx requires go1.21 to install successfully.
